Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -464,6 +467,17 @@ - (nullable NSString *)currentScreenNameForSessionReplay
                .firstObject;
 }
 
+#    pragma mark - SentryReachabilityObserver
+
+- (void)connectivityChanged:(BOOL)connected typeDescription:(nonnull NSString *)typeDescription
+{
+    if (connected) {
+        [_sessionReplay resume];
+    } else {
+        [_sessionReplay pause];
```

Comment:
`m`: That code runs on a background thread. I wonder if we need to make `resume` and `pause` thread-safe, as they might be called from other threads as well.

https://github.com/getsentry/sentry-cocoa/blob/e0abb7e49c3133b01a2ad0307415661289c2d99e/Sources/Sentry/SentryReachability.m#L200-L201

Reply:
I will use the lock for pause and resume